contracts/p1/mixins/RecollateralizationLib.sol

```diff
@@ -391,9 +414,36 @@ library RecollateralizationLibP1 {
             }
         }
 
+        // Special-case RToken
+        {
```

Comment:
just leaving a comment to remind myself to review this deeper in the morning when i'm fresh. it seems ok but but its also definitely a rather large change so i want to be sure it matches the previous logic.